R/arrow.R

```diff
@@ -0,0 +1,101 @@
+#' @param fillColor Fill color of arrow. This can be of the form "#AABBCC" or
+#'   "rgb(255,100,200)" or "yellow". Defaults to white.
+#' @param strokeColor Stroke color of arrow. This can be of the form "#AABBCC" or
+#'   "rgb(255,100,200)" or "yellow". Defaults to black.
+#'
+#' @return A dygraph with the specified trade arrow.
+#'
+#' @examples
+#' library(dygraphs)
+#' dygraph(presidents, main = "Quarterly Presidential Approval Ratings") %>%
+#'   dyArrow("1950-6-30", "Korea", direction = "down", fillColor = "red") %>%
+#'   dyArrow("1965-2-09", "Vietnam", direction = "down", fillColor = "red")
+#'
+#' dygraph(presidents, main = "Quarterly Presidential Approval Ratings") %>%
+#'   dyArrow(c("1950-6-30", "1965-2-09"),
+#'           c("Korea", "Vietnam"),
+#'           direction = "down",
+#'           fillColor = "red")
+#'
+#' @export
+dyArrow <- function(dygraph,
+                    x,
+                    text = NULL,
+                    tooltip = NULL,
+                    direction = c("up", "down", "left", "right", "ne", "se", "sw", "nw"),
+                    fillColor = "white",
+                    strokeColor = "black",
+                    series = NULL) {
+
+  # create arrows
+  if (!is.null(text) && length(x) != length(text)) {
+    stop("Length of 'x' and 'text' does not match")
+  }
+  if (!is.null(tooltip) && length(x) != length(tooltip)) {
+    stop("Length of 'x' and 'tooltip' does not match")
+  }
+
+  # validate series if specified
+  if (!is.null(series) && ! series %in% dygraph$x$attrs$labels) {
+    stop("The specified series was not found. Valid series names are: ",
+         paste(dygraph$x$attrs$labels[-1], collapse = ", "))
+  }
+
+  # default the series if necessary
+  if (is.null(series))
+    series <- dygraph$x$attrs$labels[length(dygraph$x$attrs$labels)]
+
+  fixedtz <- ifelse(is.null(dygraph$x$fixedtz), FALSE, dygraph$x$fixedtz)
+  scale <- dygraph$x$scale
+
+  direction <- match.arg(direction)
+  arrows <- lapply(seq_along(x),
+                   function(i) {
+                     if (is.null(text))
+                       arrowText <- NULL
+                     else
+                       arrowText <- text[i]
+                     if (is.null(tooltip))
+                       arrowTooltip <- NULL
+                     else
+                       arrowTooltip <- tooltip[i]
+                     list(xval = ifelse(dygraph$x$format == "date",
+                                        asISO8601Time(x[i]), x[i]),
+                          text = arrowText,
+                          tooltip = arrowTooltip,
+                          direction = direction,
+                          fillColor = fillColor,
+                          strokeColor = strokeColor,
+                          series = series,
+                          fixedtz = fixedtz,
+                          scale = scale)
+                   })
+
+  pluginName <- "Arrow"
+  if (dyHasPlugin(dygraph, pluginName)) {
```

Comment:
This behavior could just be made part of `dyPlugin` (i.e. if an a plugin name that already exists in the dygraph is added then we append rather than overwrite options). Would there be any downsides to this?